unittests/webserver/RPCServerHandlerTest.cpp

```diff
@@ -399,8 +399,9 @@ TEST_F(WebRPCServerHandlerTest, HTTPInvalidAPIVersion)
 
     (*handler)(std::move(request), session);
     std::this_thread::sleep_for(200ms);
+    rpcEngine->post(
```

Comment:
This feels and looks like a hack 🤣
Let's make MockAsyncRPCEngine::post synchronous on the caller thread:
```cpp
template <typename Fn>
bool
post(Fn&& func, [[maybe_unused]] std::string const& ip = "")
{
    using namespace boost::asio;
    io_context ioc;

    spawn(ioc, [handler = std::forward<Fn>(func), _ = make_work_guard(ioc.get_executor())](auto yield) mutable {
        handler(yield);
    });

    ioc.run();
    return true;
}
```

and perform the EXPECT_EQ on main thread as before + remove the 200ms sleeping which would no longer be needed.